dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/OrderAndFilterEventChangeLogTest.java

```diff
@@ -268,6 +268,35 @@ void shouldSortChangeLogsWhenOrderingByChangeDescAndChangesOnlyToEventFields()
         () -> assertFieldCreate("geometry", "(-11.419700, 8.103900)", changeLogs.get(4)));
   }
 
+  @Test
+  void shouldSortChangeLogsByNameWhenOrderingByChangeAndDataElementDoesNotHaveFormName()
+      throws ForbiddenException, NotFoundException {
+    EventChangeLogOperationParams params =
+        EventChangeLogOperationParams.builder().orderBy("change", SortDirection.DESC).build();
+    Event event = getEvent("pTzf9KYMk72");
+
+    updateDataValues(event, "DATAEL00001", "value00002", "value00003");
+
+    List<String> changeLogs =
+        eventChangeLogService
+            .getEventChangeLog(UID.of("pTzf9KYMk72"), params, defaultPageParams)
+            .getItems()
+            .stream()
+            .map(this::getDisplayName)
```

Comment:
Maybe you cold replace the `getDisplayName` method with with `ObjectUtils.firstNonNull`